src/wallet/asyncrpcoperation_mergetoaddress.cpp

```diff
@@ -0,0 +1,943 @@
+                    }
+                }
+                witnesses.push_back(w);
+            }
+
+            // The jsAnchor is null if this JoinSplit is at the start of a new chain
+            if (jsAnchor.IsNull()) {
+                jsAnchor = inputAnchor;
+            }
+
+            // Add spendable notes as inputs
+            std::copy(vInputNotes.begin(), vInputNotes.end(), std::back_inserter(info.notes));
+            std::copy(vInputZKeys.begin(), vInputZKeys.end(), std::back_inserter(info.zkeys));
+        }
+
+        // Accumulate change
+        jsChange = jsInputValue + info.vpub_old;
+
+        // Set vpub_new in the last joinsplit (when there are no more notes to spend)
+        if (zInputsDeque.empty()) {
+            assert(!vpubNewProcessed);
+            if (jsInputValue < vpubNewTarget) {
+                throw JSONRPCError(RPC_WALLET_ERROR,
+                                   strprintf("Insufficient funds for vpub_new %s (miners fee %s, taddr inputs %s)",
+                                             FormatMoney(vpubNewTarget), FormatMoney(minersFee), FormatMoney(t_inputs_total)));
+            }
+            info.vpub_new += vpubNewTarget; // funds flowing back to public pool
+            vpubNewProcessed = true;
+            jsChange -= vpubNewTarget;
+            // If we are merging to a t-addr, there should be no change
+            if (isToTaddr_) assert(jsChange == 0);
+        }
+
+        // create dummy output
+        info.vjsout.push_back(JSOutput()); // dummy output while we accumulate funds into a change note for vpub_new
+
+        // create output for any change
+        if (jsChange > 0) {
+            std::string outputType = "change";
+            auto jso = JSOutput(changeAddress, jsChange);
+            // If this is the final output, set the target and memo
+            if (isToZaddr_ && vpubNewProcessed) {
+                outputType = "target";
+                jso.addr = toPaymentAddress_;
+                if (!hexMemo.empty()) {
+                    jso.memo = get_memo_from_hex_string(hexMemo);
+                }
+            }
+            info.vjsout.push_back(jso);
+
+            LogPrint("zrpcunsafe", "%s: generating note for %s (amount=%s)\n",
+                     getId(),
+                     outputType,
+                     FormatMoney(jsChange));
+        }
+
+        obj = perform_joinsplit(info, witnesses, jsAnchor);
+
+        if (jsChange > 0) {
+            changeOutputIndex = mta_find_output(obj, 1);
+        }
+    }
+
+    // Sanity check in case changes to code block above exits loop by invoking 'break'
+    assert(zInputsDeque.size() == 0);
+    assert(vpubNewProcessed);
+
+    sign_send_raw_transaction(obj);
+    return true;
+}
+
+
+/**
+ * Sign and send a raw transaction.
+ * Raw transaction as hex string should be in object field "rawtxn"
+ */
+void AsyncRPCOperation_mergetoaddress::sign_send_raw_transaction(UniValue obj)
```

Comment:
`obj` could also be passed by reference:

```suggestion
void AsyncRPCOperation_mergetoaddress::sign_send_raw_transaction(UniValue const& obj)
```

Reply:
I agree with @drgora with a comment addition though: this method is named "sign_send_raw_transaction" so I'm expecting the input parameter to be a "raw_transaction" ... why we do pass the whole "UniValue" object to pull the transaction from ?

Reply:
And of course use the form `qualifier -> type -> name` (eg. `const UniValue& obj`)